jdaviz/configs/mosviz/mixins.py

```diff
@@ -0,0 +1,56 @@
+__all__ = ['Matched2dSpectrumMixin']
+
+
+class Matched2dSpectrumMixin:
+    _during_limit_sync = False
+
+    def _setup_xaxes_linking(self):
+        spec1d = self.app.get_viewer(self._default_spectrum_viewer_reference_name)
+        spec2d = self.app.get_viewer(self._default_spectrum_2d_viewer_reference_name)
+        for k in ('x_min', 'x_max'):
+            spec1d.state.add_callback(k, self._on_spec1d_x_limits_changed)
+            spec2d.state.add_callback(k, self._on_spec2d_x_limits_changed)
+
+    def _on_spec1d_x_limits_changed(self, *args):
+        if self._during_limit_sync:
+            # prevent feedback-loop
+            return
+        spec1d = self.app.get_viewer(self._default_spectrum_viewer_reference_name)
+        spec2d = self.app.get_viewer(self._default_spectrum_2d_viewer_reference_name)
+        if spec1d.state.reference_data is None or spec2d.state.reference_data is None:
+            return
+
+        spec1d_limits = (spec1d.state.x_min, spec1d.state.x_max)
+        if spec1d.state.x_display_unit == 'pix':
```

Comment:
Coverage is complaining about this logic route. Will this be covered as part of the unit conversion work?

Reply:
either then or wavelength calibration work.